### PR TITLE
Add completed_with_errors info

### PR DESCRIPTION
**Pull request by JumasJM**
- added string type completedWithErrors info for workflows with values true/false
- added list of completedWithErrorsTasks for storing specific task models
- updated json_data archive table with same variable of type string and WorkflowModelSummary

Type: Improvement
JIRA: DEP-524

### CODE REVIEW

File: common/src/main/java/com/netflix/conductor/common/run/Workflow.java [hunk 1]
```diff
@@ -126,6 +126,8 @@ public boolean isSuccessful() {
     @ProtoField(id = 25)
     private Set<String> failedTaskNames = new HashSet<>();
 
+    private String completedWithErrors;
```
Comment (marosmars) [inline common/src/main/java/com/netflix/conductor/common/run/Workflow.java:129]: why string and not Boolean ?

Comment (JumasJM) [reply common/src/main/java/com/netflix/conductor/common/run/Workflow.java]: changed to Boolean

File: core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java [hunk 2]
```diff
@@ -209,9 +210,16 @@ private DeciderOutcome decide(final WorkflowModel workflow, List<TaskModel> preS
                     outcome.tasksToBeUpdated.add(pendingTask);
                 } else {
                     pendingTask.setStatus(COMPLETED_WITH_ERRORS);
+                    workflow.setCompletedWithErrors(true);
                 }
             }
 
+            if (pendingTask.getTaskType().equals(TASK_TYPE_SUB_WORKFLOW)
```
Comment (marosmars) [inline core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java:217]: should we support this for other task types too ? besides subworkflow task types

Comment (JumasJM) [reply core/src/main/java/com/netflix/conductor/core/execution/DeciderService.java]: done

**Review by marosmars — APPROVED** [review]

**Pull request merged**